main/src/ca/uwaterloo/flix/language/errors/TypeError.scala

```diff
@@ -48,17 +48,75 @@ object TypeError {
          |
          |${code(loc, "unable to generalize the type scheme.")}
          |
+         |The declared type does not match the inferred type:
+         |
          |  Declared: ${cyan(FormatScheme.formatSchemeWithoutConstraints(declared))}
          |  Inferred: ${magenta(FormatScheme.formatSchemeWithoutConstraints(inferred))}
          |""".stripMargin
     }
 
-    /**
-      * Returns a formatted string with helpful suggestions.
-      */
+    def explain(formatter: Formatter): Option[String] = Some(
+      """The declared type is more polymorphic than the inferred type.
+        |
+        |Typically:
+        |
+        |  (a) the declared type signature is incorrect, or
+        |  (b) the expression body is incorrect.
+        |""".stripMargin)
+  }
+
+  /**
+    * Effect Generalization Error.
+    *
+    * @param declared the declared effect.
+    * @param inferred the inferred effect.
+    * @param loc      the location where the error occurred.
+    */
+  case class EffectGeneralizationError(declared: Type, inferred: Type, loc: SourceLocation) extends TypeError {
+    def summary: String = s"The inferred effect '${FormatEff.formatEff(inferred)}' cannot be generalized to '${FormatEff.formatEff(declared)}'."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.format)}
+         |>> The inferred effect: '${red(FormatEff.formatEff(inferred))}' cannot be generalized to '${red(FormatEff.formatEff(declared))}'.
+         |
+         |${code(loc, "unable to generalize the effect.")}
```

Comment:
Generalization error still *feels* like it's intended for a machine. Personally, I think
`The inferred effect '${FormatEff.formatEff(inferred)}' cannot be generalized to '${FormatEff.formatEff(declared)}'.`
is informative if you know exactly what type generalization is but for new programmers is gibberish.
Maybe the `explain` function could describe this concept a bit more in natural language so the user can learn what the concept means. I could imagine that as soon as you know what generalization errors are then the current message is better.
Same applies to `GeneralizationError` obviously.

Reply:
Agreed. Not sure yet how to deal with it.